python/lsst/daf/butler/core/dimensions/_records.py

```diff
@@ -353,6 +388,8 @@ def from_simple(cls, simple: SerializedDimensionRecord,
         if (reg := "region") in rec:
             encoded = bytes.fromhex(rec[reg])
             rec[reg] = lsst.sphgeom.Region.decode(encoded)
+        if (hsh := "hash") in rec:
+            rec[hsh] = bytes.fromhex(rec[hsh].decode())
```

Comment:
FWIW, my solution to this on DM-30332 was 6e9ed39064ff4f949e36569063323a3639465eec, uses a pydantic validator to do the conversion instead.  I can't tell from a quick glance which is better, so let's go with your version, which has been more thoroughly tested, and we can come back to it if necessary later.